### PR TITLE
dcos-commons 0.2.3 integration

### PR DESCRIPTION
**Pull request by gabrielhartmann**

### CODE REVIEW

**Comment by gabrielhartmann** [comment]
Retest this please

**Comment by gabrielhartmann** [comment]
retest please

**Comment by gabrielhartmann** [comment]
@mesosphere-ci: Retest please

**Comment by gabrielhartmann** [comment]
Retest please

File: cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraContainer.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package com.mesosphere.dcos.cassandra.common.tasks;
+
+import org.apache.mesos.Protos;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+
+/**
+ * Created by gabriel on 6/8/16.
```
Comment (mohitsoni) [inline cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraContainer.java]: Please add meaningful comment.

File: cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraDaemonTask.java [hunk 2]
```diff
@@ -414,206 +63,161 @@ private final CassandraConfig updateConfig(CassandraDaemonStatus status) {
...
+                                              Optional<String> message) {
+        return CassandraDaemonStatus.create(getStatusBuilder(state, message)
+            .setData(CassandraData.createDaemonStatusData(mode).getBytes())
+            .build());
+    }
 
+    public String getVolumePath() {
+        return TaskUtils.getVolumePaths(
+            getTaskInfo().getResourcesList())
+            .get(0);
```
Comment (mohitsoni) [inline cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraDaemonTask.java]: I want us to re-think about this. Just getting `0` always might lead us into trouble, if we ever want to support multiple volumes.

Comment (gabrielhartmann) [inline cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraDaemonTask.java]: This is a complicated topic and not part of this PR.

File: cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraTask.java [hunk 3]
```diff
@@ -133,549 +104,347 @@ public CassandraTask deserialize(byte[] bytes)
...
+        if (!volumeMode.equals(VolumeRequirement.VolumeMode.NONE)) {
+            if (volumeType.equals(VolumeRequirement.VolumeType.MOUNT)) {
+                builder.addResources(ResourceUtils.getDesiredMountVolume(role, principal, diskMb, CassandraConfig.VOLUME_PATH));
+            } else {
+                builder.addResources(ResourceUtils.getDesiredRootVolume(role, principal, diskMb, CassandraConfig.VOLUME_PATH));
+            }
+        }
+
+        if (!ports.isEmpty()) {
+            builder.addResources(createPorts(ports, role, principal));
```
Comment (gabrielhartmann) [inline cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraTask.java]: Bug here.  Should be:
```
            builder.addResources(ResourceUtils.getDesiredRanges(role, principal, "ports", Algorithms.createRanges(ports)));
```
Already fixed.  Found with test.  Will push shortly.
